integration/database.go

```diff
@@ -0,0 +1,75 @@
+package integration
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os/exec"
+	"testing"
+)
+
+// PostgresDBConfig models a test Postgres database
+type PostgresDBConfig struct {
+	Host            string
+	Port            int
+	DBUser          string
+	DBName          string
+	DBPassword      string
+	migrateFunction func(sql.DB) error
+}
+
+// Reset drops all the tables in a test database and regenerates them by
+// running migrations. If a migrationFunction has not been specified, then the
+// tables are dropped but not regenerated
+func (cfg PostgresDBConfig) Reset(t *testing.T) {
+	db, err := sql.Open("postgres", cfg.GetDSN())
+	if err != nil {
+		t.Fatalf("unable to connect to %s database: %v", cfg.DBName, err)
+	}
+	defer db.Close()
+	resetQuery := "DROP SCHEMA public CASCADE;" +
+		"CREATE SCHEMA public;" +
+		"GRANT ALL ON SCHEMA public TO postgres;" +
+		"GRANT ALL ON SCHEMA public TO public;"
+	// drop all the tables in the test database
+	if _, err := db.Exec(resetQuery); err != nil {
+		t.Fatalf("unable to drop tables in %s database: %v", cfg.DBName, err)
+	}
+	// run migrations if a migration function has exists
+	if cfg.migrateFunction != nil {
+		if err := cfg.migrateFunction(*db); err != nil {
+			t.Fatalf("unable to migrate %s database: %v", cfg.DBName, err)
+		}
+	}
+}
+
+// RunAsContainer uses the Postgres configuration to run a Postgres Docker
+// container locally
+func (cfg PostgresDBConfig) RunAsContainer(containerName string) func() {
```

Comment:
hmm, I think it's fine to have it take in a `testing.T` because that implies that it should be called on a per-test basis, which it should.

If you wanted to be fancy, you could have two versions of the method -- one that returns an `error` and one that takes in a `testing.T`. I'm not opinionated about that and don't have the interest to come up with names for each :)